src/NHibernate/Async/Criterion/QueryOver.cs

```diff
@@ -0,0 +1,69 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+using NHibernate.Criterion.Lambda;
+using NHibernate.Engine;
+using NHibernate.Impl;
+using NHibernate.SqlCommand;
+using NHibernate.Transform;
+
+namespace NHibernate.Criterion
+{
+	using System.Threading.Tasks;
+	public abstract partial class QueryOver<TRoot> : QueryOver, IQueryOver<TRoot>
+	{
+
+		private async Task<IList<TRoot>> ListAsync()
+		{
+			return await (criteria.ListAsync<TRoot>()).ConfigureAwait(false);
```

Comment:
This [other question](http://stackoverflow.com/q/41736312/1178314) is similar. 

In the case of an API having some not implemented members, for me it does not make much sens to delay the exception to the next await by putting it in a task. I would rather have it fail as early as possible.

(I would tend to do that for any cases indeed. If we fail to build the task, should we really emulate an async failure instead of letting the exception flow synchronously? I am thinking about [this](https://github.com/nhibernate/nhibernate-core/pull/588/commits/0e04aaed99bb8b09017bc9769ef04a7470fc95f8#diff-8746f99e800bd6d8671212406664f151R55). Personally I would not put all those try catch for converting exceptions to failed task.)